pyccel/codegen/printing/ccode.py

```diff
@@ -1134,35 +1181,51 @@ def formatted_args_to_printf(args_format, args, end):
             code += formatted_args_to_printf(args_format, args, end)
         return code
 
-    def find_in_dtype_registry(self, dtype):
+    def find_in_type_registry(self, dtype):
         """
-        Find the corresponding C dtype in the dtype_registry.
+        Find the corresponding C dtype in the dtype_registry or the STC container type.
 
-        Find the corresponding C dtype in the dtype_registry.
-        Raise PYCCEL_RESTRICTION_TODO if not found.
+        This function searches for the corresponding C data type in the `dtype_registry`.
+        If the provided dtype is a container (like `HomogeneousSetType` or
+        `HomogeneousListType`), it identifies the scalar type of the container and returns
+        the appropriate type for the `STC` container.
+        It raises a `PYCCEL_RESTRICTION_TODO` error if the dtype is not found in the registry.
 
         Parameters
         ----------
-        dtype : DataType
-            The data type of the expression.
+        dtype : DataType or ContainerType
```

Comment:
`DataType` was replaced/renamed in #1756. It is now called `PyccelType`. Types like `HomogeneousListType` also inherit from `PyccelType` so there is no need to add `ContainerType` here

Reply:
updated 8b4fa11b2c6dce02e30c77cef4d60da0d58e9315